File: src/scripts/convert_excel_to_json.py
import pandas as pd
import json

# Lire le fichier Excel
excel_file = './store/docs/cholera-db.xlsx'
df = pd.read_excel(excel_file)

# Convertir en JSON
json_data = df.to_json(orient='records', indent=2)

# Écrire le résultat dans un fichier JSON
output_file = './store/docs/cholera-db.json'
with open(output_file, 'w') as f:
    f.write(json_data)

print(f"Conversion terminée. Le fichier JSON a été créé à {output_file}")

File: src/scripts/convert-excel-to-json.js
const XLSX = require('xlsx');
const fs = require('fs');
const path = require('path');

// Chemin du fichier Excel et du fichier de sortie JSON
const excelFile = path.join(__dirname, 'store', 'docs', 'cholera-db.xlsx');
const outputFile = path.join(__dirname, 'store', 'docs', 'cholera-db.json');

// Lire le fichier Excel
const workbook = XLSX.readFile(excelFile);
const sheetName = workbook.SheetNames[0];
const worksheet = workbook.Sheets[sheetName];

// Convertir la feuille en JSON
const jsonData = XLSX.utils.sheet_to_json(worksheet, { defval: null });

// Écrire le JSON dans un fichier
fs.writeFileSync(outputFile, JSON.stringify(jsonData, null, 2), 'utf8');

console.log(`Conversion terminée. Le fichier JSON a été créé à ${outputFile}`);
